osu.Game/Online/Spectator/SpectatorClient.cs

```diff
@@ -168,7 +184,20 @@ public void BeginPlaying(GameplayState state, Score score)
             });
         }
 
-        public void SendFrames(FrameDataBundle data) => lastSend = SendFramesInternal(data);
+        public void HandleFrame(ReplayFrame frame) => Schedule(() =>
+        {
+            if (!IsPlaying)
+            {
+                Logger.Log($"Frames arrived at {nameof(SpectatorClient)} outside of gameplay scope and will be ignored.");
```

Comment:
I've left this logging and returning for now, but was hoping to make this throw instead. Unfortunately this causes test failures due to the aforementioned async disposal issues that this component faces. We'll probably want to look to fix that in the future as it's adding a bit of complexity around spectating flow, but not in this PR.